src/iterated_integrals.jl

```diff
@@ -0,0 +1,302 @@
+function fill_WikJ(ΔW,::Val{1},::Val{true})
+    WikJ = false .* ΔW .* ΔW
+    WikJDiagonal_iip{typeof(WikJ)}(WikJ)
+end
+
+# function WikJCommute_oop(ΔW)
+function fill_WikJ(ΔW,::Val{2},::Val{false})
+    WikJCommute_oop()
+end
+
+# function WikJCommute_iip(ΔW)
+function fill_WikJ(ΔW,::Val{2},::Val{true})
+    WikJ = false .* ΔW .* ΔW'
+    WikJCommute_iip{typeof(WikJ)}(WikJ)
+end
+
+# function WikJGeneral_oop(ΔW)
+function fill_WikJ(ΔW,::Val{3},::Val{false})
+    m = length(ΔW)
+    M = m*(m-1)/2
+    m_seq = Array{Int}(undef, M, 2)
+    k = 1
+    for i in 1:length(ΔW)
+      for j in i+1:length(ΔW)
+        m_seq[k,1] = i
+        m_seq[k,2] = j
+        k += 1
+      end
+    end
+    WikJGeneral_oop(m_seq)
+end
+
+# function WikJGeneral_iip(ΔW)
+function fill_WikJ(ΔW,::Val{3},::Val{true})
+    WikJ = false .* ΔW .* ΔW'
+    WikJ2 = false .* ΔW .* ΔW'
+    WikJ3 = false .* ΔW .* ΔW'
+    m = length(ΔW)
+    M = m*(m-1)/2
+    m_seq = Array{Int}(undef, M, 2)
+    k = 1
+    for i in 1:length(ΔW)
+      for j in i+1:length(ΔW)
+        m_seq[k,1] = i
+        m_seq[k,2] = j
+        k += 1
+      end
+    end
+    vec_ζ = false .* vec(ΔW)
+    vec_η = false .* vec(ΔW)
+    Gp₁ = false .* Array{eltype(ΔW)}(undef, M)
+    Gp₂ = false .* Array{eltype(ΔW)}(undef, M)
+    Aᵢ = false .* vec(ΔW)
+    WikJGeneral_iip{eltype(ΔW), typeof(WikJ)}(WikJ, WikJ2, WikJ3, m_seq, vec_ζ, vec_η, Gp₁, Gp₂, Aᵢ)
+end
+
+# fill_WikJ(ΔW,::Val{1},::Val{false}) = WikJDiagonal_oop(ΔW)
+# fill_WikJ(ΔW,::Val{1},::Val{true}) = WikJDiagonal_iip(ΔW)
+# fill_WikJ(ΔW,::Val{2},::Val{false}) = WikJCommute_oop(ΔW)
+# fill_WikJ(ΔW,::Val{2},::Val{true}) = WikJCommute_iip(ΔW)
+# fill_WikJ(ΔW,::Val{3},::Val{false}) = WikJGeneral_oop(ΔW)
+# fill_WikJ(ΔW,::Val{3},::Val{true}) = WikJGeneral_iip(ΔW)
+
+"""
+
+    get_iterated_I!(dW, Wik::AbstractWikJ, C=1)
+
+This function calculates WikJ, a mxm Array for a m dimensional general noise problem, which is a approximation
+to the second order iterated integrals.
+
+WikJDiagonal and WikJCommute use the properties of respective noises to simplify the calculations.
+While the calculation for General Noise case is taken from section 4 of [SDELab: A Package for solving stochastic differential
+equations in MATLAB](https://doi.org/10.1016/j.cam.2006.05.037) and SDELAB2(https://github.com/tonyshardlow/SDELAB2)
+which is the Implementation of SDELab in Julia.
+```math
+    𝒜ᵖ = (Iₘ² - Pₘ)Kₘᵀ Δt/(2π) √(𝑎ₚ) √(Σ∞) Gp₁
+```
+
+```math
+    √(Σ∞) = (Σ∞ + 2αIₘ)/(√2 * (1 + α))
+```
+
+let the combined operators be,
+```math
+    F = Kₘ(Iₘ² - Pₘ)(Iₘ ⨂ W(Δt)W(Δt)ᵀ)(Iₘ² - Pₘ)Kₘᵀ
+```
+
+```math
+    Σ∞ = 2Iₘ + (2/Δt)F
+```
+
+See the paper for further details of specific operators.
+Here we've only shown in which order these are implemented in this code.
+
+From above we can see:
+
+```math
+    Δt/(2π) √(𝑎ₚ) √(Σ∞) Gp₁ = Δt/π √(𝑎ₚ) (√(Σ∞)/2 Gp₁)
+```
+
+```math
+    Oper2(Gp₁) = (√(Σ∞)/2 Gp₁) = (Iₘ/√2 + F/(√2 * (1 + α) * Δt))(Gp₁)
```

Comment:
what's Oper2?

Reply:
I just named it so that I won't have to write the whole left part again.

Reply:
But what is it? What operation is it? Is it in the code?

Reply:
No. Just explanation purposes.